### PR TITLE
Update emulator definitions version to latest available from chrome infra

### PR DESCRIPTION
**Pull request by reidbaker**
Update to the latest definitions of emulator configurations.  (except x86/api28 see b/379736755)
Specifically to past  https://crrev.com/c/5908894 which added `gpu_mode: "swangle_indirect"` to the api 35 proto. This should help with emulator stability on versions past android 13. 
Related to  b/371020223
Research notes / my findings for how this system works at go/flutter-android-ci-emulator
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by matanlurey — APPROVED** [review]

**Comment by reidbaker** [comment]
Requires bot configuration change

**Comment by reidbaker** [comment]
Led started build here https://ci.chromium.org/ui/p/flutter/builders/staging.shadow/Linux%20linux_android_emulator_opengles_tests_34/10/overview

`led get-builder 'luci.flutter.staging:Linux linux_android_emulator_opengles_tests_34' | led edit -pa git_ref='refs/pull/56313/head' | led edit -pa git_url='https://github.com/flutter/engine' | led launch`

**Comment by reidbaker** [comment]
The [ubuntu 20](https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20linux_android_emulator_skia_tests/3672/overview) and [ubuntu 22](https://ci.chromium.org/ui/p/flutter/builders/staging.shadow/Linux%20Staging%20Engine%20Drone/185/overview) builds both have avd that is working but the 22 had a golden test fail.

**Comment by matanlurey** [comment]
> The [ubuntu 20](https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20linux_android_emulator_skia_tests/3672/overview) and [ubuntu 22](https://ci.chromium.org/ui/p/flutter/builders/staging.shadow/Linux%20Staging%20Engine%20Drone/185/overview) builds both have avd that is working but the 22 had a golden test fail.

My guess is its because the latter never had OpenGLES tests that were passing on CI (see <https://flutter-engine-gold.skia.org/detail?grouping=name%3DExternalTextureTests_testMediaSurface%26source_type%3Dflutter-engine&digest=be38dad3170e770a17f62ca51ed9a2bf> for an example.

@gaaclarke for FYI since Jonah is OOO today.

**Review by gmackall — APPROVED** [review]

**Comment by reidbaker** [comment]
Failing test log has `D1118 12:26:51.107593  378102 main-common-ui.c:393] GPU emulation enabled using 'swiftshader_indirect' mode` so it is not using `swangle`. 

My guess is that this is an x86 limitation.

File: ci/builders/linux_android_emulator.json [hunk 1]
```diff
@@ -139,6 +139,7 @@
                         },
                         {
                             "dependency": "avd_cipd_version",
+                            # Do not update build. x86 emulators do not support swangle
                             "version": "build_id:8735216702926189361"
```
[the remark this answers is not in the document]
Comment (matanlurey) [reply ci/builders/linux_android_emulator.json]: You can always leave a dummy key, i.e. `"_comment"` (I do this in a few of the files).

**Comment by reidbaker** [comment]
From https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8730897896447642385/+/u/start_avd/Start_Android_emulator__android_28_google_apis_x86.textpb_/l_execution_details we can see that the recipe change to not call xvfb is present. 
```
Executing command [
  '/b/s/w/ir/cipd_bin_packages/vpython3',
  '/b/s/w/ir/cache/avd/src/tools/android/avd/avd.py',
...
  ```
  
  from https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20Engine%20Drone/2910565/infra we can see the machine is a x64 machine and it is a linux machine without the ubuntu 22 requirement. (also there are no ubuntu 22 machines in the try pool) 
  
 ```
INFO    | Android emulator version 35.1.21.0 (build_id 12325540) (CL:N/A)
...
D1118 13:16:45.722594   47772 main-common-ui.c:393] GPU emulation enabled using 'swiftshader_indirect' mode
```
 from https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8730897896447642385/+/u/start_avd/Start_Android_emulator__android_28_google_apis_x86.textpb_/raw_io.output_text 
Shows we are using swift renderer which we would expect given that I reverted the build_id update. 


Contents at build id [8735216702926189361](https://chrome-infra-packages.appspot.com/p/chromium/tools/android/avd/linux-amd64/+/build_id:8735216702926189361) 
```
# Copyright 2024 The Chromium Authors
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

# Configuration for an Android Pie (API 28) AVD on google_apis on x86

emulator_package {
  package_name: "chromium/third_party/android_sdk/public/emulator"
  version: "XS__KK27esI_4qZkc_Ybb_ooYjPfcIQ7DDOPUZGYW5wC"  # 35.1.21 (Stable)
  dest_path: "android_28_google_apis_x86"
}

system_image_package {
  package_name: "chromium/third_party/android_sdk/public/system-images/android-28/google_apis/x86"
  version: "n5ghWmdJtOyffzCTuQwcIHCz34tTB6Cac9gJGU5pCMAC"  # r12, PSR1.180720.122
  dest_path: "android_28_google_apis_x86"
}
system_image_name: "system-images;android-28;google_apis;x86"

avd_package {
  package_name: "chromium/third_party/android_sdk/public/avds/android-28/google_apis/x86"
  # Created in https://ci.chromium.org/ui/b/8735716928436784161
  # Patched gmscore version 24.26.32 in https://crrev.com/c/5735581
  version: "4ZAU1LOIDDo3l2FcZ7vd4Ry9A4xJg_XfygyhoZpGGDYC"
  dest_path: "android_28_google_apis_x86"
}
avd_name: "android_28_google_apis_x86"
``` 

 This does not have the swangle renderer enabled. 
Edit: It looks like the file [android_28_google_apis_x86.textpb](https://chromium.googlesource.com/chromium/src.git/+/HEAD/tools/android/avd/proto/android_28_google_apis_x86.textpb) in question has not been updated since october 1 2024 where the emulator version was updated to 35.1.21.

**Review by gmackall — APPROVED** [review]

**Comment by reidbaker** [comment]
> Golden file changes have been found for this pull request. Click [here to view and triage](https://flutter-engine-gold.skia.org/cl/github/56313) (e.g. because this is an intentional change).
> 
> If you are still iterating on this change and are not ready to resolve the images on the Flutter Gold dashboard, consider marking this PR as a draft pull request above. You will still be able to view image results on the dashboard, commenting will be silenced, and the check will not try to resolve itself until marked ready for review.
> 
> _Changes reported for pull request #56313 at sha [dc32feb](https://github.com/flutter/engine/commit/dc32feba7b47ed931981594cd95a5a56e45743fd)_

I think this is because I am changing the emulator render. Approving goldens

**Pull request merged**